pkg/placement/selectors.go

```diff
@@ -72,3 +75,50 @@ func GetKeys(m map[string]string) []string {
 	}
 	return keys
 }
+
+// when an object is updated, we iterate over all placements and update placement-resolutions that
+// are affected by the update. Every affected placement decision is then queued to be synced.
+func (c *Controller) updateDecisions(obj runtime.Object) error {
+	placements, err := c.listPlacements()
+	if err != nil {
+		return err
+	}
+
+	objMR := obj.(mrObject)
+
+	for _, item := range placements {
+		placement, err := runtimeObjectToPlacement(item)
+		if err != nil {
+			return err
+		}
+
+		matchedSome := c.testObject(objMR, placement.Spec.Downsync)
+		if !matchedSome {
+			// if previously selected, remove
+			// TODO: optimize
+			if resolutionUpdated := c.placementResolver.RemoveObject(placement.GetName(), obj); resolutionUpdated {
+				// enqueue placement-decision to be synced since object was removed from its placement's resolution
+				c.enqueuePlacementDecision(placement.GetName())
+				c.logger.V(4).Info("enqueued placement-decision %s for syncing due to the removal of an " +
+					"object to its resolution")
+			}
+			continue
+		}
+
+		// obj is selected by placement, update the placement decision resolver
+		resolutionUpdated, err := c.placementResolver.NoteObject(placement.GetName(), obj)
+		if err != nil {
+			return fmt.Errorf("failed to update resolution for placement %s for object %v: %v",
+				placement.GetName(), util.GenerateObjectInfoString(obj), err)
+		}
+
+		if resolutionUpdated {
+			// enqueue placement-decision to be synced since an object was added to its placement's resolution
+			c.enqueuePlacementDecision(placement.GetName())
+			c.logger.V(4).Info("enqueued placement-decision %s for syncing due to the addition of an " +
```

Comment:
Info takes message and key-value pairs, not format and arguments. You forgot the PlacementDecision's name.

Reply:
Or you could just include the GVR and the `klog.KSObj` as two key-value pairs.

Reply:
Fixed the problem (similar to previous comment). Regarding the object identification using `util.GenerateObjectInfoString`, for now im keeping it unified with the rest of the code. This can be a part of the Key refactoring since that function lives there for some reason.